### PR TITLE
Working xenial install

### PR DESCRIPTION
**Pull request by iynere**
@heug & i just went over this—i spun up a new server install w/ it & got green realitycheck

obviously we shouldn't merge w/o changelogging for a server release, but i think we should aim for 2.15 or 2.16 here if possible (not that it would affect existing installations—but just to keep things organized)

@junejung doesn't seem to be anything in Jira for this work—can we get something created for tracking purposes ?

## changes
- `linux-image-extra` package naming conventions
- docker package (`trusty` to `xenial`)
- virtual network interface name changes b/w `trusty` & `xenial` (`eth0` doesn't exist in xenial)
- different way of starting nomad service
- bumped all amis to latest `ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server` images in all regions

### CODE REVIEW

File: templates/services_user_data.tpl [hunk 1]
```diff
@@ -15,7 +15,7 @@ apt-get update && apt-get -y upgrade
 echo "--------------------------------------------"
 echo "       Setting Private IP"
 echo "--------------------------------------------"
-export PRIVATE_IP="$(/sbin/ifconfig eth0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}')"
+export PRIVATE_IP="$(/sbin/ifconfig ens3 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}')"
```
Comment (GERey) [inline templates/services_user_data.tpl]: does it always end up being in ens3 for these installs?

Comment (iynere) [reply templates/services_user_data.tpl]: ya as far as i know @GERey

Comment (iynere) [reply templates/services_user_data.tpl]: or else `enp0s3`: https://askubuntu.com/questions/889880/cannot-find-device-eth0#889882
or `ens33`: https://itzgeek.com/how-tos/mini-howtos/change-default-network-name-ens33-to-old-eth0-on-ubuntu-16-04.html

we should replace this with a shell command to programatically grab the right network interface

Comment (iynere) [reply templates/services_user_data.tpl]: something like `lshw -C network`

**Comment by GERey** [comment]
I guess I'm also a little confused about the ask here. This would change anybody who pulls upstream regularly without noticing over to 16.04. I do agree that maybe it would be wise to switch over to that after some heads up, but feels like it should be handled in a module rather than making the changes directly to the defaults. @mqarty thoughts?

**Comment by iynere** [comment]
do you think folks actually pull in changes to this repo ? @GERey my impression is, they don't do that, because they all modify our Terraform code beyond recognition... if that's the case, though, then we'd want to be intentional about this. that said, we _do_ need a Xenial version that we can make available to customers ASAP.

**Comment by OompahLoompah** [comment]
This work is duplicated in #114

**Pull request closed without merging**